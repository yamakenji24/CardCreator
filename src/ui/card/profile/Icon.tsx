interface Props {
  name: string | null;
  icon: string;
}

export const Icon = ({name, icon}: Props) => {
  if(!name) return null;

  return (
    <div className="mr-0 ml-24 text-gray-100 flex">
      <div className="h-8 w-8 mt-1 mr-2">
        <img
          src={"/_next/image?url=%2F"+icon+".svg&w=64&q=75"} 
          alt={name} 
          width="25" height="25"
        />
      </div>
      <p>{name}</p>
    </div>
  );
};
