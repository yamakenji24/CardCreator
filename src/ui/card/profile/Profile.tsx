import { Icon } from "./Icon";
import { ProfileType } from '../../../utils/getProfile';
import Image from 'next/image';

export const Profile = ({
  name, picture, twitter, provider, email, blog
}: ProfileType): JSX.Element => {
  return (
    <div className="py-2 pl-12">
      <div className="flex flex-row">
        <Image　src={picture}　alt={name}　className="rounded-full"　width="120" height="120"/>
        <div className="flex flex-col">
          <p className="text-3xl ml-24 mb-3 -mt-4 font-bold text-white">{name}</p>
          <Icon name={name} icon={provider}/>
          <Icon name={twitter} icon="twitter" />
          <Icon name={email} icon="mail"/>
          <Icon name={blog} icon="archive" />
        </div>
      </div>
    </div>
  );
};
