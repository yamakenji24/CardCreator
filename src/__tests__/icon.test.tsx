import { render } from '@testing-library/react';

import { Icon } from '../ui/card/profile/Icon';

describe("<Icon />", () => {
  test("render icon img and text snapshot", () => {
    const {asFragment} = render(<Icon name="testing" icon="logo" />)
    expect(asFragment()).toMatchSnapshot()
  }),
  test("render null when name=null", () => {
    const wrapper = render(<Icon name={null} icon=''/>)
    expect(wrapper.container.innerHTML).toHaveLength(0);
  })
})