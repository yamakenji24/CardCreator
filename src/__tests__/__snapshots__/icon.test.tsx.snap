// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> render icon img and text snapshot 1`] = `
<DocumentFragment>
  <div
    class="mr-0 ml-24 text-gray-100 flex"
  >
    <div
      class="h-8 w-8 mt-1 mr-2"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,<svg width=\\"20\\" height=\\"20\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="testing"
          decoding="async"
          src="/_next/image?url=%2Flogo.svg&w=320&q=75"
          srcset="/_next/image?url=%2Flogo.svg&w=320&q=75 1x"
          style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <p>
      testing
    </p>
  </div>
</DocumentFragment>
`;
